src/dataset-builder/ConceptSearch.ts

```diff
@@ -0,0 +1,150 @@
+};
+
+const DebouncedTextInput = withDebouncedChange(TextInput);
+export const ConceptSearch = (props: ConceptSearchProps) => {
+  const { initialSearch, domainOption, onCancel, onCommit, onOpenHierarchy, actionText, datasetId, initialCart } =
+    props;
+  const [search, setSearch] = useState<string>(initialSearch);
+  const [cart, setCart] = useState<Concept[]>(initialCart);
+  const [concepts, searchConcepts] = useLoadedData<GetConceptsResponse>();
+
+  useEffect(() => {
+    void searchConcepts(() => {
+      return DataRepo().dataset(datasetId).searchConcepts(domainOption.root, search);
+    });
+  }, [search, datasetId, domainOption.root, searchConcepts]);
+  const tableLeftPadding = { paddingLeft: '2rem' };
+
+  return h(Fragment, [
+    h(BuilderPageHeader, [
+      h2({ style: { display: 'flex', alignItems: 'center' } }, [
+        h(
+          Link,
+          {
+            onClick: onCancel,
+            'aria-label': 'cancel',
+          },
+          [icon('left-circle-filled', { size: 32 })]
+        ),
+        div({ style: { marginLeft: 15 } }, [domainOption.category]),
+      ]),
+      div({ style: { position: 'relative' } }, [
+        h(DebouncedTextInput, {
+          onChange: (value: string) => {
+            setSearch(value);
+          },
+          value: search,
+          placeholder: 'Search',
+          type: 'search',
+          style: {
+            borderRadius: 25,
+            borderColor: colors.dark(0.2),
+            width: '100%',
+            maxWidth: 575,
+            height: '3rem',
+            marginRight: 20,
+            paddingLeft: 40,
+          },
+        }),
+        icon('search', { size: 18, style: { position: 'absolute', left: 15, top: '50%', marginTop: -9 } }),
+      ]),
+      concepts.status === 'Ready'
+        ? h(SimpleTable, {
+            'aria-label': 'concept search results',
+            underRowKey: 'underRow',
+            rowStyle: {
+              backgroundColor: 'white',
+              ...tableLeftPadding,
+            },
+            headerRowStyle: {
+              ...tableHeaderStyle,
+              ...tableLeftPadding,
+              marginTop: '1rem',
+            },
+            cellStyle: {
+              paddingTop: 10,
+              paddingBottom: 10,
+            },
+            columns: [
+              { header: strong(['Concept name']), width: 710, key: 'name' },
+              { header: strong(['Concept ID']), width: 195, key: 'id' },
+              { header: strong(['Code']), width: 195, key: 'code' },
+              { header: strong(['Roll-up count']), width: 205, key: 'count' },
+              { width: 100, key: 'hierarchy' },
+            ],
+            rows: _.map((concept) => {
+              const [label, iconName]: [string, IconId] = (() => {
+                if (_.contains(concept, cart)) {
+                  return ['remove', 'minus-circle-red'];
+                }
+                return ['add', 'plus-circle-filled'];
+              })();
```

Comment:
Optional nit: I think you can rework this function to be more terse by using a ternary.
```suggestion
              const [label, iconName]: [string, IconId] = (() => _.contains(concept, cart) ? ['remove', 'minus-circle-red'] : ['add', 'plus-circle-filled'])();
```

Reply:
Sure, and then I can get rid of the IIFE too. Can't recall why I wrote it this way before, the code was probably more complex then.